src/FiniteVolumeTools.cpp

```diff
@@ -1673,6 +1673,44 @@ void GeneralizedBarycentricCoordinates(
 			
 		}
 		
+		//Use nearest neighbor weighting if there are too many edges.  This is necessary to avoid numerical instability.  
+		
+		if( iEdges > 30 ){
```

Comment:
Should I also test with a coarse CS and a fine RLL mesh to trigger this case? What is the resolution combo that you tested for this edge case?

Reply:
This only occurs for one specific land source mesh with triangular elements where there is a mesh node that is shared by 720 source faces (it occurs near the south pole, all the other nodes are fine). The dual mesh face that surrounds this node then contains 720 edges, which leads to extreme ill-condition of the algorithm that is used to generated the barycentric coordinates.  There are no RLL or CS meshes of any resolution that will lead to this problem.  Note that the bilinear weighting does not face this problem because faces that contain more than 4 edges are triangulated.

I can send you this mesh if you want to test it.  Let me know.

Reply:
Thanks for sending me the relevant meshes.